### PR TITLE
adding logs to task.go, system.go and redis.go  files

### PR DESCRIPTION
**Pull request by lakkundipooja**
adding logs to task.go, system.go and redis.go  files

Resolves : https://github.com/ODIM-Project/ODIM/issues/428

### CODE REVIEW

File: lib-persistence-manager/persistencemgr/redis.go [hunk 1]
```diff
@@ -326,9 +335,10 @@ func (p *ConnPool) Update(table, key string, data interface{}) (string, *errors.
 	if err != nil {
 		return "", errors.PackError(errors.UndefinedErrorType, "Write to DB in json form failed: "+err.Error())
 	}
+
 	writePool := (*redis.Pool)(atomic.LoadPointer((*unsafe.Pointer)(unsafe.Pointer(&p.WritePool))))
 	if writePool == nil {
-		return "", errors.PackError(errors.UndefinedErrorType, "write DB pool is nil ")
+		return "", errors.PackError(errors.UndefinedErrorType, "Update : Writepool is nil ")
```
Comment (Bharath-KKB) [inline lib-persistence-manager/persistencemgr/redis.go:341]: add log  here

File: lib-persistence-manager/persistencemgr/redis.go [hunk 2]
```diff
@@ -326,9 +335,10 @@ func (p *ConnPool) Update(table, key string, data interface{}) (string, *errors.
 	if err != nil {
 		return "", errors.PackError(errors.UndefinedErrorType, "Write to DB in json form failed: "+err.Error())
```
Comment (Bharath-KKB) [inline lib-persistence-manager/persistencemgr/redis.go:336]: add log here

File: lib-persistence-manager/persistencemgr/redis.go [hunk 3]
```diff
@@ -280,7 +289,7 @@ func (c *Config) Connection() (*ConnPool, *errors.Error) {
 func (p *ConnPool) Create(table, key string, data interface{}) *errors.Error {
 	writePool := (*redis.Pool)(atomic.LoadPointer((*unsafe.Pointer)(unsafe.Pointer(&p.WritePool))))
 	if writePool == nil {
-		return errors.PackError(errors.UndefinedErrorType, "WritePool is nil ")
+		return errors.PackError(errors.UndefinedErrorType, "Create : WritePool is nil ")
```
Comment (Bharath-KKB) [inline lib-persistence-manager/persistencemgr/redis.go:292]: add log here

File: lib-persistence-manager/persistencemgr/redis.go [hunk 4]
```diff
@@ -411,20 +421,24 @@ func (p *ConnPool) GetAllDetails(table string) ([]string, *errors.Error) {
 //Delete data entry
 // Read takes "key" sting as input which acts as a unique ID to delete specific data from DB
 func (p *ConnPool) Delete(table, key string) *errors.Error {
+
 	writePool := (*redis.Pool)(atomic.LoadPointer((*unsafe.Pointer)(unsafe.Pointer(&p.WritePool))))
+
 	if writePool == nil {
 		return errors.PackError(errors.UndefinedErrorType, "error while trying to delete data: WritePool is nil ")
```
Comment (Bharath-KKB) [inline lib-persistence-manager/persistencemgr/redis.go:428]: add log here

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
add a few logs as per comment

File: lib-persistence-manager/persistencemgr/redis.go (hunk 1 above)
Comment (lakkundipooja) [reply lib-persistence-manager/persistencemgr/redis.go:346]: added logs.

File: lib-persistence-manager/persistencemgr/redis.go (hunk 2 above)
Comment (lakkundipooja) [reply lib-persistence-manager/persistencemgr/redis.go:339]: added logs

File: lib-persistence-manager/persistencemgr/redis.go (hunk 3 above)
Comment (lakkundipooja) [reply lib-persistence-manager/persistencemgr/redis.go:294]: added logs

File: lib-persistence-manager/persistencemgr/redis.go (hunk 4 above)
Comment (lakkundipooja) [reply lib-persistence-manager/persistencemgr/redis.go:434]: addressed

File: lib-persistence-manager/persistencemgr/redis.go [hunk 5]
```diff
@@ -108,7 +108,7 @@ func resetDBWriteConection(dbFlag DbType) {
 			config := getInMemoryDBConfig()
 			log.Debug("getInMemoryDBConfig: ", config)
 			currentMasterIP, currentMasterPort := GetCurrentMasterHostPort(config)
-			log.Info("InMemory currentMaster IP: "+currentMasterIP, "CurrentMaster Port: "+currentMasterPort, "inMemDBConnPool.MasterIP: "+inMemDBConnPool.MasterIP)
+			log.Info("InMemory currentMaster IP: "+currentMasterIP, ", CurrentMaster Port: "+currentMasterPort, " ,InMemDBConnPool.MasterIP: "+inMemDBConnPool.MasterIP)
```
Comment (a-ajith) [inline lib-persistence-manager/persistencemgr/redis.go:111]: `+currentMasterIP,`  shouldn't be `+currentMasterIP+`
same with `+currentMasterPort,`

**Review by Bharath-KKB — APPROVED** [review]

File: lib-persistence-manager/persistencemgr/redis.go [hunk 6]
```diff
@@ -177,8 +177,12 @@ func GetDBConnection(dbFlag DbType) (*ConnPool, *errors.Error) {
 	case InMemory:
 		// In this case this function return in-memory db connection pool
 		if inMemDBConnPool == nil || inMemDBConnPool.ReadPool == nil {
+			log.Info("GetDBConnection : inMemDBConnPool OR inMemDBConnPool.ReadPool is nil")
 			config := getInMemoryDBConfig()
 			inMemDBConnPool, err = config.Connection()
+			if err != nil {
+				log.Error("error while trying to get Inmemory Readpool connection : " + err.Error())
+			}
```
Comment (a-ajith) [inline lib-persistence-manager/persistencemgr/redis.go:185]: shouldn't we return here?

File: lib-persistence-manager/persistencemgr/redis.go [hunk 7]
```diff
@@ -193,6 +197,9 @@ func GetDBConnection(dbFlag DbType) (*ConnPool, *errors.Error) {
 		if onDiskDBConnPool == nil || onDiskDBConnPool.ReadPool == nil {
 			config := getOnDiskDBConfig()
 			onDiskDBConnPool, err = config.Connection()
+			if err != nil {
+				log.Error("error while trying to get Ondisk Readpool connection : " + err.Error())
+			}
```
Comment (a-ajith) [inline lib-persistence-manager/persistencemgr/redis.go:202]: please refer to the above comment

File: lib-persistence-manager/persistencemgr/redis.go [hunk 8]
```diff
@@ -254,6 +261,7 @@ func (c *Config) Connection() (*ConnPool, *errors.Error) {
 	//Check if any connection error occured
 	if err != nil {
 		if errs, aye := isDbConnectError(err); aye {
+			log.Error("error while trying to get Readpool connection : " + errs.Error())
```
Comment (a-ajith) [inline lib-persistence-manager/persistencemgr/redis.go:264]: please remove "error" from the log message

File: lib-persistence-manager/persistencemgr/redis.go (hunk 6 above)
Comment (Bharath-KKB) [reply lib-persistence-manager/persistencemgr/redis.go:185]: There was something Hemantha had said about this. Dont remember but dont change right now

File: lib-persistence-manager/persistencemgr/redis.go (hunk 7 above)
Comment (Bharath-KKB) [reply lib-persistence-manager/persistencemgr/redis.go:202]: As above dont change right now

**Review by Bharath-KKB — COMMENTED** [review]
do not address Arjuns comment on returning on error.

**Review by a-ajith — APPROVED** [review]

**Review by bharath-b-hpe — APPROVED** [review]

File: lib-persistence-manager/persistencemgr/redis.go (hunk 8 above)
Comment (lakkundipooja) [reply lib-persistence-manager/persistencemgr/redis.go:264]: will be corrected in upcoming PRs.

File: lib-persistence-manager/persistencemgr/redis.go (hunk 5 above)
Comment (lakkundipooja) [reply lib-persistence-manager/persistencemgr/redis.go]: addressed.

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**